xwiki-pro-macros-test/xwiki-pro-macros-test-pageobjects/src/main/java/com/xwiki/pro/test/po/generic/UserListMacro.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.pro.test.po.generic;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.xwiki.test.ui.po.BaseElement;
+
+/**
+ * Represents a UserList macro and provides access to its attributes.
+ *
+ * @version $Id$
+ * @since 1.28
+ */
+public class UserListMacro extends BaseElement
+{
+    private final WebElement userList;
+
+    public UserListMacro(WebElement userList)
+    {
+        this.userList = userList;
+    }
+
+    public int getUserCount()
+    {
+        return userList.findElements(By.cssSelector("tbody tr")).size();
+    }
+
+    public WebElement getUserRow(int userIndex)
+    {
+        return userList.findElements(By.cssSelector("tbody tr")).get(userIndex);
+    }
+
+    public int getUserPropertiesCount(int userIndex)
+    {
+        return getUserRow(userIndex).findElements(By.tagName("td")).size();
+    }
+
+    public List<WebElement> getUserProperties(int userIndex)
+    {
+        return getUserRow(userIndex).findElements(By.tagName("td"));
+    }
+
+    public List<String> getUserPropertiesText(int userIndex)
+    {
+        return getUserProperties(userIndex).stream().map(WebElement::getText).collect(Collectors.toList());
+    }
+
+    public List<String> getUserPropertyTypes(int userIndex)
+    {
+        List<WebElement> tds = getUserProperties(userIndex);
+        List<String> types = new ArrayList<>();
+
+        for (WebElement td : tds) {
+            String classAttr = td.getAttribute("class");
+            if (classAttr != null && classAttr.startsWith("xwiki-userlist-user-")) {
+                types.add(classAttr.substring("xwiki-userlist-user-".length()));
+            } else {
+                types.add("");
+            }
+        }
+
+        return types;
+    }
+
+    public boolean isEmailLinkValid(int userIndex, String expectedEmail)
+    {
+        WebElement emailTd = getUserProperties(userIndex).get(3);
```

Comment:
I would not access properties based on their index. This can break easily on changes 
I would use the class instead, since it contains a particularity